### ISSUE TITLE
new architecture of the Python-MapReduce-Xtandem

### ISSUE THREAD

**Issue by baimingze** [baimingze/project1#6]
### Main idea
1. why MapReduce, not Spark? we follow the mr-xtandem, implement the MapReduce version at first.
2.  From Top to Down(by Sangzhe): follow the same MapRduce design of mr-xtandem, implement the multiple map/reduce task, as follow(make a fake process method and some fake results):
   ![archtecture design1](https://cloud.githubusercontent.com/assets/3970400/9479867/5bf779a2-4b77-11e5-86a1-d05281d36158.jpg)
3. From Down to Top(by Mingze): find a way to wrap the original c++ class files as a library, which can be call by cython.

**Comment by ypriverol** [comment]
@baimingze it would be interesting to test first the mr-xtandem running and see how it works for a real experiment and the comparison with a normal xTandem search. In my view, the first thing you need to do to start a software project is to evaluate the previous implementation. We can try using a public experiment in PRIDE or GPMDB. I can design the search experiment, then run the Xtandem in a node, then the hadoop implementation, and then you can move on!!! 
What do you think?

**Comment by baimingze** [comment]
Great suggestion, we can test them on the EBI Hadoop cluster.

**Comment by ypriverol** [comment]
@baimingze let me know when you have ready the code of the haddop implementation we can run a simple project with a couple of files.

**Comment by sangzhe** [comment]
1.write hadoop-streaming program for xtandem in python language.
2.xtandem is wrapped as python extension .
Do I get the point ?But I'm still confused about something more detailed

**Comment by baimingze** [comment]
> 1.write hadoop-streaming program for xtandem in python language.

YES, based on the figure above, do your best to implement the figure, with some null function/object (just use pass in the function);

> 2.xtandem is wrapped as python extension .

I will do this part. In your implementation, all things are in Python.

> Do I get the point ?But I'm still confused about something more detailed

You have got the point, all questions are welcome here.
